src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -78,29 +85,27 @@ private JavadocPropertiesGenerator() {
      * TokenTypes.properties generator entry point.
      * @param args the command line arguments
      * @throws CheckstyleException if parser or lexer failed or if there is an IO problem
-     * @throws ParseException if the command line can not be passed
      **/
-    public static void main(String... args)
-            throws CheckstyleException, ParseException {
-        final CommandLine commandLine = parseCli(args);
-        if (commandLine.getArgList().size() == 1) {
-            final File inputFile = new File(commandLine.getArgList().get(0));
-            final File outputFile = new File(commandLine.getOptionValue(OPTION_DEST_FILE));
-            writePropertiesFile(inputFile, outputFile);
+    public static void main(String... args) throws CheckstyleException {
+        final JavadocPropertiesGenerator generator = new JavadocPropertiesGenerator();
+        final CommandLine cmd = new CommandLine(generator).setUsageHelpWidth(USAGE_HELP_WIDTH);
+        try {
+            final ParseResult parseResult = cmd.parseArgs(args);
+            if (!CommandLine.printHelpIfRequested(parseResult)) {
+                generator.writePropertiesFile();
+            }
         }
-        else {
-            printUsage();
+        catch (ParameterException ex) {
+            System.err.println(ex.getMessage());
+            ex.getCommandLine().usage(System.err);
```

Comment:
The reason is that error reporting for invalid user input is much nicer: instead of a stack trace, the user will see an error message indicating what they did wrong and the usage help message showing what options are available.
You are correct that this changes the behaviour but I hope you agree this is slightly more user-friendly.

For errors other than invalid user input, the exception is passed up the chain as before.